src/components/registration/steps.jsx

```diff
@@ -494,7 +515,7 @@ module.exports = {
                                  tipContent={formatMessage({id: 'registration.nameStepTooltip'})} />
                     </p>
                     <Card>
-                        <Form onValidSubmit={this.onValidSubmit}>
+                        <Form onValidSubmit={this.onValidSubmit} onSubmit={this.onSubmit}>
```

Comment:
Now this can be `onValidSubmit={this.props.onNextStep}` and the `onValidSubmit` method can go away.